src/popup/components/Modals/ClaimGiftCard.vue

```diff
@@ -0,0 +1,406 @@
+<template>
+  <Modal
+    from-bottom
+    body-without-padding-bottom
+    has-close-button
+    class="claim-gift-card"
+    @close="reject"
+  >
+    <div class="header text-heading-2">
+      {{ $t('modals.claimGiftCard.title') }}
+    </div>
+    <template v-if="step === STEPS.redeemPartially || (step === STEPS.redeemFull && !isCardEmpty)">
+      <CheckCircleIcon class="check-circle-icon" />
+      <div class="redeemed-amount">
+        <span>{{ `${amount} AE` }}</span>
+        <span class="gray">
+          {{ $t('modals.claimGiftCard.redeemedBy', [`(${currencyFormatted})`]) }}
+        </span>
+      </div>
+    </template>
+    <DetailsItem>
+      <template #label>
+        <AccountSelector
+          v-if="step !== STEPS.redeemPartially && step !== STEPS.redeemFull"
+          v-model="recipientId"
+          :options="aeAccountsSelectOptions"
+        />
+        <AccountItem
+          v-else
+          :address="recipientId"
+          :protocol="PROTOCOL_AETERNITY"
+        />
+      </template>
+    </DetailsItem>
+    <div class="redeem-balance">
+      <span v-if="step !== STEPS.initial || !isCardEmpty">
+        {{ $t('modals.claimGiftCard.balance') }}
+      </span>
+      <BalanceInfo
+        :balance="balance.toNumber()"
+        :protocol="PROTOCOL_AETERNITY"
+        :class="{
+          gray: balance.toNumber() === 0,
+        }"
+      />
+    </div>
+    <span
+      v-if="isCardEmpty"
+      class="already-redeemed"
+    >
+      {{ $t('modals.claimGiftCard.alreadyRedeemed') }}
+    </span>
+    <TransferSendAmount
+      v-if="step === STEPS.form"
+      v-model="amount"
+      :errors="errors"
+      readonly
+      :protocol="PROTOCOL_AETERNITY"
+      custom-label="Amount to redeem"
+      :validation-rules="{
+        max_redeem: max.toString(),
+      }"
+    >
+      <template #label-after>
+        <BtnPlain
+          class="max-button"
+          :class="{ chosen: amount.toString() === max.toString() }"
+          @click="setMaxAmount"
+        >
+          {{ $t('common.max') }}
+        </BtnPlain>
+      </template>
+    </TransferSendAmount>
+    <template #footer>
+      <div class="buttons">
+        <BtnMain
+          v-if="step === STEPS.initial"
+          variant="muted"
+          :text="$t('modals.claimGiftCard.redeemPartially')"
+          :disabled="loading"
+          @click="step = STEPS.form"
+        />
+        <BtnMain
+          :icon="loading ? AnimatedSpinner : null"
+          :text="mainButtonText"
+          :disabled="isDisabled || loading"
+          @click="handleMainButtonClick"
```

Comment:
```suggestion
          @click="handleMainButtonClick()"
```
This avoids passing the click event as the 1st function argument.